compiler/rustc_query_impl/src/on_disk_cache.rs

```diff
@@ -502,7 +502,7 @@ impl<'a, 'tcx> CacheDecoder<'a, 'tcx> {
                 // expansion, so we use `import_source_files` to ensure that the foreign
                 // source files are actually imported before we call `source_file_by_stable_id`.
                 if stable_id.cnum != LOCAL_CRATE {
-                    self.tcx.cstore_untracked().import_source_files(self.tcx.sess, stable_id.cnum);
+                    self.tcx.cstore(()).import_source_files(self.tcx.sess, stable_id.cnum);
```

Comment:
This needs to use cstore_untracked to prevent every query result being decoded even if green and all dependent crates are green.